internal/network/none_platform_node_ips_allocation.go

```diff
@@ -0,0 +1,325 @@
+			return fmt.Sprintf("{host-id: %s, ip %s}", n.hostId, n.ip.String())
+		}), ","),
+		strings.Join(lo.Map(c.prohibitedNetworks, func(h *hostNetwork, _ int) string {
+			return fmt.Sprintf("{host-id: %s, network %s}", h.hostId, h.ipNet.String())
+		}), ","))
+}
+
+type nodeIpAllocator struct {
+	candidates []*hostCandidates
+	clusterId  strfmt.UUID
+	log        logrus.FieldLogger
+}
+
+func (n *nodeIpAllocator) sort(allocations []*allocationSolution) {
+	sort.SliceStable(allocations, func(i, j int) bool {
+		return allocations[i].routeMetric < allocations[j].routeMetric
+	})
+}
+
+// Allocate node IPs.  Since the node IPs are dependent on each other, then all of them must be allocated
+// through this allocation procedure.  It tries to find an allocation when no conflict exists.
+func (n *nodeIpAllocator) allocate(index int, parentContext *collisionDetector) (map[strfmt.UUID]*NodeIpAllocation, []error) {
+	if index == len(n.candidates) {
+		return make(map[strfmt.UUID]*NodeIpAllocation), nil
+	}
+	var (
+		errs                []error
+		allocationSolutions []*allocationSolution
+		h                   = n.candidates[index]
+	)
+	for i := range h.eligibleNodeIps {
+		c := h.eligibleNodeIps[i]
+		localCollisionContext := newCollisionDetectorForNodeIp(h, c)
+		n.log.Debugf("local collision context: %s", localCollisionContext)
+		if collisionErrs := localCollisionContext.detectCollisions(parentContext); len(collisionErrs) > 0 {
+			errs = append(errs, collisionErrs...)
+			continue
+		}
+		result, childErrs := n.allocate(index+1, parentContext.merge(localCollisionContext))
+		if len(childErrs) > 0 {
+			errs = append(errs, childErrs...)
+			continue
+		}
+		result[h.hostId] = &NodeIpAllocation{
+			NodeIp: c.nodeIp.String(),
+			HintIp: c.ipNet.IP.String(),
+			Cidr:   c.ipNet.String(),
+		}
+		allocationSolutions = append(allocationSolutions, &allocationSolution{
+			routeMetric:       c.routeMetric,
+			nodeIpAllocations: result,
+		})
+	}
+	if len(allocationSolutions) > 0 {
+		n.sort(allocationSolutions)
+		return allocationSolutions[0].nodeIpAllocations, nil
+	}
+	return nil, errs
+}
+
+func (n *nodeIpAllocator) allocateNodeIps() (map[strfmt.UUID]*NodeIpAllocation, error) {
+	result, errs := n.allocate(0, newCollisionDetector())
+	uniqErrs := lo.UniqBy(errs, func(e error) string { return e.Error() })
+	err := stderrors.Join(uniqErrs...)
+	if err != nil {
+		n.log.WithError(err).Errorf("failed to allocate master node ips for cluster %s", n.clusterId)
+		return nil, err
+	}
+	return result, nil
+}
+
+func GenerateNonePlatformAddressAllocation(cluster *common.Cluster, log logrus.FieldLogger) (map[strfmt.UUID]*NodeIpAllocation, error) {
+	addressFamilies, err := GetClusterAddressFamilies(cluster)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed tp get address families for cluster %s", lo.FromPtr(cluster.ID))
+	}
+	isIpv4 := lo.Contains(addressFamilies, IPv4)
+	isIpv6 := lo.Contains(addressFamilies, IPv6)
```

Comment:
should not we check ipv6 too? I don't see we the place we create host candidates for ipv6?
What i miss?

Reply:
I think we do not support dual stack with IPv6 primary

Reply:
We have a validation that stops it? I mean not dual stack installation but ipv6 only when there are ipv4 ips on the node

Reply:
In this case `GetClusterAddressFamilies` will return only IPv6.
It relies on ServiceNetworks and ClusterNetworks